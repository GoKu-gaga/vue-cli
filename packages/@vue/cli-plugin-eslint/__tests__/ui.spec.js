const configDescriptor = require('../ui/configDescriptor')
const { getEslintConfigName, getDefaultValue } = configDescriptor

describe('getEslintConfigName', () => {
  describe('for "extend" of string type', () => {
    it('returns null if it doesn\'t extend vue plugin config', () => {
      expect(getEslintConfigName({
        extends: 'eslint:recommended'
      })).toBe(null)
    })

    it('returns vue config used', () => {
      expect(getEslintConfigName({
        extends: 'plugin:vue/recommended'
      })).toBe('plugin:vue/recommended')
    })
  })

  describe('for "extend" of array type', () => {
    it('returns null if it doesn\'t extend vue plugin config', () => {
      expect(getEslintConfigName({
        extends: ['eslint:recommended', 'standard']
      })).toBe(null)
    })

    it('returns vue config used', () => {
      expect(getEslintConfigName({
        extends: ['eslint:recommended', 'plugin:vue/recommended']
      })).toBe('plugin:vue/recommended')
    })
  })
})

describe('getDefaultValue', () => {
  const getResult = (config, ruleCategory) => {
    const rule = {
      meta: {
        docs: {
          category: ruleCategory
        }
      }
    }

    const data = {
      eslint: {
        extends: config
      }
    }

    return getDefaultValue(rule, data)
  }

  it('returns "ERROR" value if the rule belongs to the selected configuration', () => {
    expect(getResult('plugin:vue/base', 'base')).toBe('error')
    expect(getResult('plugin:vue/essential', 'base')).toBe('error')
    expect(getResult('plugin:vue/essential', 'essential')).toBe('error')
    expect(getResult('plugin:vue/strongly-recommended', 'base')).toBe('error')
    expect(getResult('plugin:vue/strongly-recommended', 'essential')).toBe('error')
    expect(getResult('plugin:vue/strongly-recommended', 'strongly-recommended')).toBe('error')
    expect(getResult('plugin:vue/recommended', 'base')).toBe('error')
    expect(getResult('plugin:vue/recommended', 'essential')).toBe('error')
    expect(getResult('plugin:vue/recommended', 'strongly-recommended')).toBe('error')
    expect(getResult('plugin:vue/recommended', 'recommended')).toBe('error')
  })

  it('returns "OFF" value if the rule doesn\'t belong to the selected configuration', () => {
    expect(getResult('plugin:vue/base', 'essential')).toBe('off')
    expect(getResult('plugin:vue/base', 'strongly-recommended')).toBe('off')
    expect(getResult('plugin:vue/base', 'recommended')).toBe('off')
    expect(getResult('plugin:vue/essential', 'strongly-recommended')).toBe('off')
    expect(getResult('plugin:vue/essential', 'recommended')).toBe('off')
    expect(getResult('plugin:vue/strongly-recommended', 'recommended')).toBe('off')
    expect(getResult('plugin:vue/base', undefined)).toBe('off')
    expect(getResult('plugin:vue/essential', undefined)).toBe('off')
    expect(getResult('plugin:vue/strongly-recommended', undefined)).toBe('off')
    expect(getResult('plugin:vue/recommended', undefined)).toBe('off')
  })
})
