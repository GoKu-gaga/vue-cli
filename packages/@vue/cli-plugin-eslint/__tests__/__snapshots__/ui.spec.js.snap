// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "name": "org.vue.eslint.config.eslint.setting.off",
        "value": "\\"off\\"",
      },
      Object {
        "name": "org.vue.eslint.config.eslint.setting.error",
        "value": "\\"error\\"",
      },
      Object {
        "name": "org.vue.eslint.config.eslint.setting.warning",
        "value": "\\"warning\\"",
      },
    ],
    "default": "\\"error\\"",
    "description": "Dolor description",
    "group": "org.vue.eslint.config.eslint.groups.base",
    "link": "http://test.com/dolor",
    "message": "vue/dolor",
    "name": "vue/dolor",
    "type": "list",
    "value": undefined,
  },
  Object {
    "choices": Array [
      Object {
        "name": "org.vue.eslint.config.eslint.setting.off",
        "value": "\\"off\\"",
      },
      Object {
        "name": "org.vue.eslint.config.eslint.setting.error",
        "value": "\\"error\\"",
      },
      Object {
        "name": "org.vue.eslint.config.eslint.setting.warning",
        "value": "\\"warning\\"",
      },
    ],
    "default": "\\"error\\"",
    "description": "Ipsum description",
    "group": "org.vue.eslint.config.eslint.groups.recommended",
    "link": "http://test.com/ipsum",
    "message": "vue/ipsum",
    "name": "vue/ipsum",
    "type": "list",
    "value": "\\"warning\\"",
  },
  Object {
    "choices": Array [
      Object {
        "name": "org.vue.eslint.config.eslint.setting.off",
        "value": "\\"off\\"",
      },
      Object {
        "name": "org.vue.eslint.config.eslint.setting.error",
        "value": "\\"error\\"",
      },
      Object {
        "name": "org.vue.eslint.config.eslint.setting.warning",
        "value": "\\"warning\\"",
      },
      Object {
        "name": "org.vue.eslint.config.eslint.setting.custom",
        "value": "[\\"error\\",[\\"asd\\"]]",
      },
    ],
    "default": "\\"off\\"",
    "description": "Lorem description",
    "group": "org.vue.eslint.config.eslint.groups.uncategorized",
    "link": "http://test.com/lorem",
    "message": "vue/lorem",
    "name": "vue/lorem",
    "type": "list",
    "value": "[\\"error\\",[\\"asd\\"]]",
  },
]
`;
